### PR TITLE
[fix bug 1109132] Implement Firefox Hello FTUE

### PR DESCRIPTION
**Pull request by alexgibson**
This has already been tested by both Firefox and Web QA pretty heavily on demo2, but still needs code review. If you're testing using 35 beta, please make sure it is up to date. All of the features used here are also in the latest Nightly.

To test locally, open `about:config`

1.) set `loop.gettingStarted.url` = `http://127.0.0.1:8000/%LOCALE%/firefox/%VERSION%/hello/start/`
2.) set `loop.gettingstarted.seen` = `false`
3.) whitelist your local server to use UiTour API by creating a pref called `browser.uitour.testingOrigins` and give it a value of `http://127.0.0.1:8000`
4.) restart the browser
5.) Click the hello icon in the browser to open the Hello panel.

You should now see a blue button that says “Get Started”. Clicking this will open the tour and if all the above steps have worked you should see the FTE page on demo and the first door-hanger pointing at “Start a conversation”

Note if you want to retake the tour without resetting `loop.gettingstarted.seen`, the tour is also available via the ‘Tour’ link in the Hello panel gear menu

### CODE REVIEW

**Comment by alexgibson** [comment]
@jpetto r?

File: bedrock/firefox/tests/test_base.py [hunk 1]
```diff
@@ -1065,3 +1065,39 @@ def test_whatsnew_search_tour_oldversion(self):
...
+
+class TestHelloStartView(TestCase):
+    def setUp(self):
+        self.user_agent = ('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:35.0) '
+                      'Gecko/20100101 Firefox/35.0')
+        self.url = reverse('firefox.hello.start', args=['35.0'])
+
+    @override_settings(DEV=True)
+    @patch.dict(product_details.firefox_versions,
+                LATEST_FIREFOX_VERSION='16.0')
```
Comment (jpetto) [inline bedrock/firefox/tests/test_base.py]: Are these decorators needed?

Comment (alexgibson) [inline bedrock/firefox/tests/test_base.py]: I'm not too sure? I'm just copying how other tests in this file have been done :smile:

File: media/css/firefox/hello/start.less [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    .font-size(24px);
+    font-style: italic;
+    counter-reset: steps-counter;
+
+    li {
+        position: relative;
+        margin: (@baseLine / 2) 0;
+        padding: 0;
+        line-height: 1.5;
+        padding-left: 2em;
```
Comment (jpetto) [inline media/css/firefox/hello/start.less]: Nit - two padding declarations.

File: media/css/firefox/hello/start.less [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+.html-rtl {
+    .spacer {
+        float: left;
+    }
+
+    ol {
+
+        li {
```
Comment (jpetto) [inline media/css/firefox/hello/start.less]: Nit - unnecessary nesting.

File: media/js/firefox/hello/start.js [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+                gaTrack(['_trackEvent', '/hello/start interactions', 'tour', 'URLCopied-Tour']);
+                break;
+            case 'Loop:RoomURLEmailed':
+                tourStep = 'shared';
+                showTourStep();
+                // track user has clicked email button
+                gaTrack(['_trackEvent', '/hello/start interactions', 'tour', 'URLEmailed-Tour']);
+            }
+
+        }, function () {
```
Comment (jpetto) [inline media/js/firefox/hello/start.js]: Is this second param/function required? Remove it if not?

Comment (alexgibson) [inline media/js/firefox/hello/start.js]: Even though we're not doing anything with this callback, I'd actually like to keep it in here as a reminder to assist debugging locally in the future.

File: bedrock/firefox/tests/test_base.py (hunk 1 above)
Comment (alexgibson) [inline bedrock/firefox/tests/test_base.py]: These are most certainly not needed, removing :)

Comment (jpetto) [inline bedrock/firefox/tests/test_base.py]: That's exactly my methodology for adding tests. :grin:

**Comment by jpetto** [comment]
A couple tiny, non-blocking nits, but otherwise looks great! r+ :tv:

**Comment by alexgibson** [comment]
Thanks @jpetto! - once #2631 is merged a bit later today, I'll rebase this branch this and add the links to the product page.

**Comment by alexgibson** [comment]
@jpetto - rebased and updated links to the product page

**Pull request merged**